src/sentry/sentry_metrics/last_seen_updater.py

```diff
@@ -32,8 +84,26 @@ def close(self) -> None:
         pass
 
     def join(self, timeout: Optional[float] = None) -> None:
-        logger.info(f"{self.__counter} messages processed")
-        self.__counter = 0
+        keys_to_pass_to_update = len(self.__seen_ints)
+        logger.debug(f"{keys_to_pass_to_update} unique keys seen")
+        self.__metrics.incr(
+            "last_seen_updater.unique_update_candidate_keys", amount=keys_to_pass_to_update
+        )
+        with self.__metrics.timer("last_seen_updater.postgres_time"):
+            update_count = _update_stale_last_seen(self.__seen_ints)
+        self.__metrics.incr("last_seen_updater.updated_rows_count", amount=update_count)
+        logger.debug(f"{update_count} keys updated")
+        self.__seen_ints = set()
+
+
+def retrieve_db_read_keys(message: Message[KafkaPayload]) -> Set[int]:
+    parsed_message = json.loads(message.payload.value, use_rapid_json=True)
```

Comment:
Please catch invalid json messages, log an error (so it will appear in Sentry) and discard the message.